plugins/modules/docker_compose2.py

```diff
@@ -0,0 +1,866 @@
+        state: absent
+
+    - community.docker.docker_compose:
+        project_name: flask
+        definition:
+            db:
+              image: postgres
+            web:
+              build: "{{ playbook_dir }}/flask"
+              command: "python manage.py runserver 0.0.0.0:8000"
+              volumes:
+                - "{{ playbook_dir }}/flask:/code"
+              ports:
+                - "8000:8000"
+              links:
+                - db
+      register: output
+
+    - ansible.builtin.debug:
+        var: output
+
+    - ansible.builtin.assert:
+        that:
+          - "'started' in containers['flask_web_1'] | default([])"
+          - "'started' in containers['flask_db_1'] | default([])"
+'''
+
+RETURN = '''
+stdout:
+  description:
+  - The stdout from docker-compose.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: str
+stderr:
+  description:
+  - The stderr from docker-compose.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: str
+containers:
+  description:
+  - A dictionary mapping the various status of containers during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: complex:
+  contains:
+    container_name:
+      description: Name of the container.
+      returned: always, unless when C(docker-compose) wasn't given the chance to run
+      type: list
+      elements: str
+      example: ["stopped", "removed"]
+volumes:
+  description:
+  - A dictionary mapping the various status of volumes during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: complex:
+  contains:
+    volume_name:
+      description: Name of the volume.
+      returned: always, unless when C(docker-compose) wasn't given the chance to run
+      type: list
+      elements: str
+      example: ["created"]
+images:
+  description:
+  - A dictionary mapping the various status of volumes during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: complex:
+  contains:
+    image_name:
+      description: Name of the image.
+      returned: always, unless when C(docker-compose) wasn't given the chance to run
+      type: list
+      elements: str
+      example: ["removed"]
+networks:
+  description:
+  - A dictionary mapping the various status of networks during C(docker-compose) operation.
+  returned: always, unless when C(docker-compose) wasn't given the chance to run
+  type: complex:
+  contains:
+    image_name:
+      description: Name of the image.
+      returned: always, unless when C(docker-compose) wasn't given the chance to run
+      type: list
+      elements: str
+      example: ["created"]
+'''
+
+
+from typing import List, Optional, Tuple, Union, Literal, Final, FrozenSet
+import re
+from collections import defaultdict
+import enum
```

Comment:
Please note that `enum` is not available in Python 2.7.

Reply:
Damn, I overlooked that this collection still supported Python 2.7. I knew I read it somewhere in #540. I'll also have to change type hint to the comment style and remove all occurrence of `/` and `*` parameters.

Reply:
Replaced with a crappy enum-like class (just a namespace of `object` instances).